src/platform_impl/windows/drop_handler.rs

```diff
@@ -77,22 +79,25 @@ impl FileDropHandler {
         this: *mut IDropTarget,
         pDataObj: *const IDataObject,
         _grfKeyState: u32,
-        _pt: *const POINTL,
+        pt: POINTL,
         pdwEffect: *mut u32,
     ) -> HRESULT {
-        use crate::event::WindowEvent::HoveredFile;
+        use crate::event::WindowEvent::DragEnter;
         let drop_handler = unsafe { Self::from_interface(this) };
-        let hdrop = unsafe {
-            Self::iterate_filenames(pDataObj, |filename| {
-                drop_handler.send_event(Event::WindowEvent {
-                    window_id: WindowId::from_raw(drop_handler.window as usize),
-                    event: HoveredFile(filename),
-                });
-            })
-        };
-        drop_handler.hovered_is_valid = hdrop.is_some();
+        let mut pt = POINT { x: pt.x, y: pt.y };
+        unsafe {
+            ScreenToClient(drop_handler.window, &mut pt);
+        }
+        let position = PhysicalPosition::new(pt.x as f64, pt.y as f64);
+        let mut paths = Vec::new();
+        let hdrop = unsafe { Self::iterate_filenames(pDataObj, |path| paths.push(path)) };
+        drop_handler.send_event(Event::WindowEvent {
+            window_id: WindowId::from_raw(drop_handler.window as usize),
+            event: DragEnter { paths, position },
+        });
+        drop_handler.enter_is_valid = hdrop.is_some();
```

Comment:
if we want to utilize it, we should use it for sedening the `Enter` as well?

Reply:
Changed it so no events are emitted if `!enter_is_valid`.

X11 has the opposite issue: if there are no valid files being dragged, it won't emit any drag start/position events but *will* emit `DragLeave`. That's a pre-existing issue (in the current master branch, it will emit `HoveredFileCancelled` without any prior `HoveredFile` events), but I'll go ahead and fix that too

Reply:
Can't we stash whether the `enter` was emitted for X11?

Reply:
Yes; it's already stored as `self.dnd.has_entered` (I'll push those changes)

Reply:
Do we actually need to do something when we have invalid drag from our perspective? Can't we early return throughout and do nothing?